core/src/main/java/org/eclipse/hono/util/MessageHelper.java

```diff
@@ -293,6 +347,27 @@ public static void annotate(final Message msg, final ResourceIdentifier resource
         MessageHelper.addAnnotation(msg, APP_PROPERTY_RESOURCE, resourceIdentifier.toString());
     }
 
+    /**
+     * Adds JMS vendor properties defined by <a href="">AMQP JMS Mapping 1.0</a> as AMQP 1.0
+     * application properties to a given message.
+     * <p>
+     * The following vendor properties are added (if a value is set):
+     * <ul>
+     * <li>{@link #JMS_VENDOR_PROPERTY_CONTENT_TYPE}</li>
+     * <li>{@link #JMS_VENDOR_PROPERTY_CONTENT_ENCODING}</li>
+     * </ul>
+     *
+     * @param msg the message to add the vendor properties to.
+     */
+    public static void addJmsVendorProperties(final Message msg) {
+        if (msg.getContentType() != null) {
```

Comment:
Do we need to check if it's empty as well ? Avoiding to set the related JMS vendor property ?